tests/epyccel/test_epyccel_complex_func.py

```diff
@@ -7,6 +7,9 @@
 import modules.complex_func as mod
 from pyccel.epyccel import epyccel
 
+ATOL = 1e-15
+RTOL = 2e-14
+
```

Comment:
I'm not sure I agree. While machines implementing IEEE standards should give equivalent results for casts and simple operations, I don't think that we can guarantee that all machines that pyccel could encounter will implement this everywhere. Personally, I would use tolerances throughout.

If we really want to check exact equality then this can be done where there is casting, and maybe where there are very simple operations. But I suspect mixing equality and tolerance is a recipe for getting them mixed up sometimes. It's harder to standardise the handling with two different setups and creates more work if the test is modified slightly. It also prevents us from creating a universal `test_results` function

Reply:
Personally I would prefer to switch from exact to approximate comparison only in the cases where this is necessary, but you are making some solid arguments here. If using approximate comparison for floats ultimately means more unit tests which are more maintainable, I must agree.